components/support/ktx/src/main/java/mozilla/components/support/ktx/android/view/View.kt

```diff
@@ -105,6 +106,25 @@ fun View.toScope(): CoroutineScope {
     return scope
 }
 
+/**
+ * Finds and returns a [View] with [predicate] as the base case, recursively.
+ */
+fun View.findViewInHierarchy(predicate: (View) -> Boolean): View? {
+    fun findView(view: View): View? {
+        if (predicate(view)) return view
+
+        if (view is ViewGroup) {
+            for (i in 0 until view.childCount) {
+                val childView = findView(view.getChildAt(i))
+                if (childView != null) return childView
+            }
+        }
+        return null
+    }
+
+    return findView(this)
```

Comment:
Is the inner `findView()` function needed? Couldn't we call `findViewInHierarchy` internally directly as `view.getChildAt(i).findViewInHierarchy(predicate)`? 🤔